classes/OccurrenceManager.php

```diff
@@ -934,25 +945,16 @@ private function setChecklistVariables($clid){
 	}
 
 	//Misc support functions
-	private function hasFullSpatialSupport(){
-		$serverStr = '';
-		if(mysqli_get_server_info($this->conn)) $serverStr = mysqli_get_server_info($this->conn);
-		else $serverStr = shell_exec('mysql -V');
-		if($serverStr){
-			if(strpos($serverStr,'MariaDB') !== false) return true;
-			else{	//db = mysql;
-				preg_match('@[0-9]+\.[0-9]+\.[0-9]+@',$serverStr,$m);
-				$mysqlVerNums = explode('.', $m[0]);
-				if($mysqlVerNums[0] > 5) return true;
-				elseif($mysqlVerNums[0] == 5){
-					if($mysqlVerNums[1] > 6) return true;
-					elseif($mysqlVerNums[1] == 6){
-						if($mysqlVerNums[2] >= 1) return true;
-					}
-				}
+	public function getMaterialSampleTypeArr(){
+		$retArr = array();
+		$sql = 'SELECT DISTINCT sampleType FROM ommaterialsample ORDER BY sampleType';
+		if($rs = $this->conn->query($sql)){
+			while($r = $rs->fetch_object()){
+				$retArr[] = $r->sampleType;
```

Comment:
This way of adding elements in PHP hurts my brain lol. This totally looks like you're reassigning the array to a string value every time in the loop.

Reply:
No issue. Just PHP syntax looking crazy from a javaScript/Java background haha.